torch/_decomp/__init__.py

```diff
@@ -263,417 +277,377 @@ def remove_decompositions(
+    aten._reshape_alias,
+    aten.rad2deg,
+    aten.rad2deg_,
+    aten.reflection_pad1d,
+    aten.reflection_pad1d_backward,
+    aten.reflection_pad2d,
+    aten.reflection_pad2d_backward,
+    aten.reflection_pad3d,
+    aten.reflection_pad3d_backward,
+    aten.replication_pad1d,
+    aten.replication_pad2d,
+    aten.replication_pad3d,
+    aten.renorm,
+    aten.renorm_,
+    aten.replication_pad2d,
+    aten.resize_as,
+    aten.roll,
+    aten.rot90,
+    aten.rrelu_with_noise,
+    aten.rrelu_with_noise_,
+    aten.rsub,
+    aten._safe_softmax,
+    aten._scaled_dot_product_flash_attention_for_cpu.default,
+    aten.select_backward,
+    aten.select_scatter,
+    aten.sgn,
+    aten.sgn_,
+    aten.sigmoid_backward,
+    aten.silu,
+    aten.silu_,
+    aten.silu_backward,
+    aten.sinc,
+    aten.sinc_,
+    aten.slice_backward,
+    aten.smooth_l1_loss,
+    aten.smooth_l1_loss_backward,
+    aten.soft_margin_loss,
+    aten.soft_margin_loss_backward,
+    aten._softmax_backward_data,
+    aten.softplus,
+    aten.softplus_backward,
+    aten.softshrink,
+    aten.special_entr,
+    aten.special_log_ndtr,
+    aten.special_xlog1py,
+    aten.split.Tensor,
+    aten.split_with_sizes_copy,
+    aten.squeeze_copy,
+    aten.squeeze.default,
+    aten.squeeze.dim,
+    aten.std,
+    aten.std_mean,
+    aten.stack,
+    aten.sum.default,
+    aten.sum.out,
+    aten.t,
+    aten.t_copy,
+    aten.take,
+    aten.tanh_backward,
+    aten.threshold,
+    aten.threshold_,
+    aten.threshold_backward,
+    aten.trace,
+    aten.transpose.int,
+    aten.transpose_copy,
+    aten.tril,
+    aten.tril_,
+    aten.triu,
+    aten.triu_,
+    aten.unbind,
+    aten.unfold_backward,
+    aten.unfold_copy,
+    aten._unsafe_index,
+    aten._unsafe_index_put,
+    aten._unsafe_masked_index,
+    aten._unsafe_masked_index_put_accumulate,
+    aten.unsafe_split.Tensor,
+    aten.unsafe_split_with_sizes,
+    aten.unsqueeze_copy,
+    aten._unsafe_view,
+    aten.upsample_linear1d,
+    aten.upsample_bilinear2d,
+    aten.upsample_trilinear3d,
+    aten.upsample_nearest2d_backward,
+    aten.view_as_complex,
+    aten.xlogy,
+    aten.xlogy_,
+    aten.zero,
+    aten.zero_,
+    aten.zeros,
+    aten.zeros_like,
+    aten._chunk_cat,
+    aten._weight_norm_interface,
+]
 
-    if not hasattr(op_overload, "_schema"):
-        return False
 
-    alias_info = len(
-        [i for i in op_overload._schema.arguments if i.alias_info is not None]
-    )
+# See NOTE [Core ATen Ops]
+#
+# list was copied from torch/_inductor/decomposition.py
+# excluding decompositions that results in prim ops
+# Resulting opset of decomposition is core aten ops
+def core_aten_decompositions() -> "CustomDecompTableMapping[OperatorBase, Callable]":
+    return CustomDecompTableMapping()
 
-    is_mutating_or_aliasing = alias_info != 0 or op_overload._schema.is_mutable
 
-    if is_mutating_or_aliasing:
-        return False
+class CustomDecompTableMapping(Dict):
```

Comment:
Hmm two things (it's a bit hard for me to understand all of the changes in this file):

(1) if we're saying that `torch.export.default_decompositions()` is the source of truth, and `torch._decomp.core_aten_decompositions()` is just an old private API where they will alias in the long term, it feels more reasonable to put all of this custom dict logic in some export files (e.g. `_export/utils.py`, or maybe some new file like `_export/default_decompositions.py`). Then we can just update `core_aten_decompositions()` to be:
```
def core_aten_decompositions():
    from _export.whatever import default_decompositions
    return default_decompositions()
```

(2) I would probably split the change into two separate PR's to mitigate risk a bit: 1 PR to add the new lazy dict logic and create `torch.export.default_decompositions()`, and a 2nd PR that updates `core_aten_decompositions()` to alias to it (which runs the tiny risk of changing behavior for people in OSS that you can't migrate directly to the new export API)